[🎨design]: 기본 페이지 Layout 구현

gayeon4679 opened the pull request:
## 📝 설명
기본 layout을 구현했습니다.

## ✅ PR 유형
- [x] CSS 등 사용자 UI 디자인 변경


## 💻 작업 내용

## 🎨 UI
![image](https://github.com/user-attachments/assets/0c945086-9226-4789-9fe4-81b5d3bbed2d)

## 🧪 레이아웃 테스트 코드
```
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import Layout from '@/pages/Layout/Layout';
export default function App() {
  return (
    <Router>
      <div>
        <Layout>
          <div></div>
        </Layout>
        {/*테스트 경로 */}
        <Routes>
          <Route path="/" />
          <Route path="/home" />
          <Route path="/mypage" />
          <Route path="/mypage/:userId/purchasehistory" />
          <Route path="/product/:productId" />
          <Route path="/product/:productId/reviews" />
          <Route path="/payment/:orderId" />
        </Routes>
      </div>
    </Router>
  );
}
```

## 💬 PR 포인트
추후 header 아이콘 위치 수정할 예정입니다!

gayeon4679 committed:
[🎨design]: 기본 페이지 Layout 구현

meowTarae commented on src/pages/Layout/Layout.tsx:
```diff
@@ -0,0 +1,41 @@
+      <div className="mb-8 mt-8">
+        <BreadcrumbAndTitle />
+      </div>
+      <div className="flex justify-between">
+        <div>{children}</div>
+        <div className="w-[12rem]">
+          <SideBar menuItems={menuItems} />
+        </div>
+      </div>
```
`{children}` 을  div로 감싸주셨는데, 혹시 이유가 있으실까요 ?

gayeon4679 replied:
children에 justify-between 적용하려고 넣었어요! sidebar 제거한다고 하면 굳이 필요하지는 않을 것 같네요

gayeon4679 replied:
div를 제거하고 수정한 내용을 커밋했습니다!

meowTarae commented on src/pages/Layout/Layout.tsx:
```diff
@@ -0,0 +1,41 @@
+      </div>
+      <div className="flex justify-between">
+        <div>{children}</div>
+        <div className="w-[12rem]">
+          <SideBar menuItems={menuItems} />
+        </div>
```
SideBar의 경우 마이페이지 내부에서만 공통적으로 사용되는 부분인 것 같습니다.
때문에 제거해주시는게 좋을 것 같아요!

gayeon4679 replied:
아... 제거하도록 하겠습니다🫡

gayeon4679 replied:
Sidebar 제거하여 수정하였습니다.

meowTarae commented on src/pages/Layout/Layout.tsx:
```diff
@@ -0,0 +1,41 @@
+  return (
+    <div className="min-h-[calc(100vh-1rem)] mt-4 mx-4 pt-4 px-20 pb-[4rem] border border-gray-300">
+      <ChevronsDown className="absolute left-1/2 transform -translate-x-1/2 top-[2rem]" />
+      <Header />
```
Header 컴포넌트의 아이콘들 전부 css의 cursor-pointer 속성을 부여해주면 좋을 것 같아요.
유저 입장에서는 마우스 커서 변환 요소도 상당히 크게 체감된답니다.

gayeon4679 replied:
넵 알겠습니다!!

gayeon4679 replied:
cursor-pointer랑 아이콘 hover 시 애니메이션까지 추가해서 수정했습니다.

meowTarae commented on src/pages/Layout/Layout.tsx:
```diff
@@ -0,0 +1,41 @@
+export default function Layout({ children }: LayoutProps) {
+  return (
+    <div className="min-h-[calc(100vh-1rem)] mt-4 mx-4 pt-4 px-20 pb-[4rem] border border-gray-300">
+      <ChevronsDown className="absolute left-1/2 transform -translate-x-1/2 top-[2rem]" />
```
여기 아래꺾쇠 아이콘도 cursor-pointer 추가 부탁 드려요!
또 마우스 Hover시 약간의 애니메이션도 적용해보면 더 좋을 것 같아요.

gayeon4679 replied:
cursor-pointer와 hover시 애니메이션 추가하여 수정하였습니다

meowTarae commented on src/pages/Layout/Layout.tsx:
```diff
@@ -0,0 +1,41 @@
+export default function Layout({ children }: LayoutProps) {
+  return (
+    <div className="min-h-[calc(100vh-1rem)] mt-4 mx-4 pt-4 px-20 pb-[4rem] border border-gray-300">
+      <ChevronsDown className="absolute left-1/2 transform -translate-x-1/2 top-[2rem]" />
```
해당 PR이 디자인이고, 약간 기능적인 부분이긴 합니다만...
모든 페이지에서 사용되는 Layout 이기 때문에, 기본적인 슬라이드 기능정도는 추가해주시면 좋을 것 같아요.

기본적인 슬라이드란, 아래꺾쇠 아이콘 클릭하면 Layout이 닫히는 애니메이션을 뜻해요!

gayeon4679 replied:
기능 추가할게요!!

gayeon4679 replied:
슬라이드 애니메이션 구현하여 수정하였습니다

meowTarae commented on src/pages/Layout/Layout.tsx:
```diff
@@ -0,0 +1,41 @@
+export default function Layout({ children }: LayoutProps) {
+  return (
+    <div className="min-h-[calc(100vh-1rem)] mt-4 mx-4 pt-4 px-20 pb-[4rem] border border-gray-300">
+      <ChevronsDown className="absolute left-1/2 transform -translate-x-1/2 top-[2rem]" />
```
추가로 그리고 상단 Nav-Bar와 조금 겹치는 것 같은데, 위로 조금 옮기는건 어떨까요 ?
마찬가지로 색상도 조금만 연하게 두면 좋을 것 같아요. Hover했을 때 원래 색상으로 돌아오는 방법도 있겠네요!

gayeon4679 replied:
좋은 생각인 것 같습니다 그렇게 수정할게요

gayeon4679 replied:
navbar 위치 조정하여 수정하였습니다

meowTarae commented on src/pages/Layout/Layout.tsx:
```diff
@@ -0,0 +1,41 @@
+          <SideBar menuItems={menuItems} />
+        </div>
+      </div>
+      <BottomNav />
```
위에 꺾쇠랑 마찬가지 이유로,  BottomNav 부분의 아이콘들도 클릭했을 때 Layout이 올라오는 애니메이션도 구현해주시면 좋을 것 같습니다.

gayeon4679 replied:
넵 수정하겠습니다!

gayeon4679 replied:
클릭 시 layout 올라오는 애니메이션 추가하여 수정하였습니다

meowTarae commented on src/pages/Layout/Layout.tsx:
```diff
@@ -0,0 +1,41 @@
+  return (
+    <div className="min-h-[calc(100vh-1rem)] mt-4 mx-4 pt-4 px-20 pb-[4rem] border border-gray-300">
+      <ChevronsDown className="absolute left-1/2 transform -translate-x-1/2 top-[2rem]" />
+      <Header />
```
Header 컴포넌트를 Layout에 적용시키니 구분이 약간 어려운 것 같아요.

'border-b border-gray-300' 를 부여해보는건 어떨까요? 테두리 색상은 편하신걸로 선택해주세요~

gayeon4679 replied:
헤더에 테두리가 없는게 디자인적으로 조금 더 괜찮은 것 같다는 생각이 드는데  테두리 없이 하는건 어떨까요??

meowTarae commented on src/pages/Layout/Layout.tsx:
```diff
@@ -0,0 +1,41 @@
+//! 메뉴아이템 url 변경
+export default function Layout({ children }: LayoutProps) {
+  return (
+    <div className="min-h-[calc(100vh-1rem)] mt-4 mx-4 pt-4 px-20 pb-[4rem] border border-gray-300">
```
테두리에 border-radius 값을 부여하면 좋을 것 같아요.

rounded-xl ? rounded-2xl ?  저는 3xl도 괜찮다고 생각해요.
적용해보시구 괜찮아보이시는걸로 선택 해주세요~

gayeon4679 replied:
넵 수정하겠습니다

gayeon4679 replied:
rounded-xl 넣어서 수정했습니다!

meowTarae commented on src/pages/Layout/Layout.tsx:
```diff
@@ -0,0 +1,41 @@
+      <ChevronsDown className="absolute left-1/2 transform -translate-x-1/2 top-[2rem]" />
+      <Header />
+      <div className="mb-8 mt-8">
+        <BreadcrumbAndTitle />
```
BreadCrumb 위치가 조금 애매한 것 같아요.
하이코딩 로고와 수직으로 맞게 떨어지면 좋을 것 같은데, 이 부분은 어떻게 생각하실까요 ?

gayeon4679 replied:
좋아요 저도 그 부분이 신경쓰여서 header에서의 margin을 없애는 방식으로 고치려고 했어요😊

gayeon4679 replied:
header에 padding을 추가하여 수정했습니다!

meowTarae reviewed (approved):
직접 만든 공통 컴포넌트들을 사용하여 바텀업 방식으로 공통 레이아웃을 제작해주셨네요.
레고를 쌓아나가는 지금의 방식이 처음이실터라 낯설고 많이 어려울 수 있었을텐데, 고생 많으셨습니다 👍

gayeon4679 committed:
[♻️refactor]:리뷰반영하여 수정

border-radius추가, ChevronsDown에 cursor-pointer과 클릭 시 애니메이션 추가, Layout 닫히는 애니메이션 추가, Layout 올라오는 애니메이션 추가
